qiskit/opflow/gradients/circuit_gradients/circuit_gradient.py

```diff
@@ -66,3 +67,32 @@ def convert(
             ValueError: If ``params`` contains a parameter not present in ``operator``.
         """
         raise NotImplementedError
+
+    @staticmethod
+    def _transpile_to_supported_operations(
+        circuit: QuantumCircuit, supported_gates: Set[str]
+    ) -> QuantumCircuit:
+        """Unroll the given circuit into a gate set for which the gradients may be computed.
+
+        Args:
+            circuit: Quantum circuit to be unrolled into supported operations.
+
+        Returns:
+            Quantum circuit which is unrolled into supported operations.
+
+        Raises:
+            QiskitError: when circuit unrolling fails.
+
+        """
+        unique_ops = set(circuit.count_ops().keys())
```

Comment:
It's faster to just call set() on the dict:

```suggestion
        unique_ops = set(circuit.count_ops())
```